src/python/pants/backend/project_info/depgraph.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import json
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from json import JSONEncoder
+from typing import Iterable, cast
+
+from pants.backend.python.target_types import PythonRequirementsField
+from pants.base.specs import AddressSpecs, MaybeEmptyDescendantAddresses
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem, LineOriented
+from pants.engine.internals.selectors import MultiGet
+from pants.engine.rules import Get, collect_rules, goal_rule, rule
+from pants.engine.target import Dependencies, DependenciesRequest, Targets
+from pants.util.meta import frozen_after_init
+
+logger = logging.getLogger(__name__)
+
+
+class Granularity(Enum):
+    FILE = "file"
+    TARGET = "target"
+    DIRECTORY = "directory"
+
+
+class GraphFormat(Enum):
+    JSON = "json"
+    DOT = "dot"
```

Comment:
I don't think pants should be in the business of creating outputs in formats not supported in python stdlib...
it would be trivial to chain/pipe the output into a different tool and have it output something the looks nice..
just my 2c.